### PR TITLE
fix: prefer_collection_literals + non_bool_condition errors

### PR DESCRIPTION
**Pull request by HelenGezahegn**
**Changes**
- deleted `prefer_collection_literals` from errors ignore list
- deleted `non_bool_condition` from errors ignore list
- fixed all violations of this lint rule

**UI**
> No visible changes

### CODE REVIEW

File: lib/chat/user_message_rows.dart [hunk 1]
```diff
@@ -129,7 +129,7 @@ buildMessageRows(context) {
                         )
                       ),
                       children: [
-                        if(dummyData[i]["isSent"]) WidgetSpan(
+                        if(dummyData[i]["isSent"] != null) WidgetSpan(
```
Comment (nayancprakash) [inline lib/chat/user_message_rows.dart:132]: does this still work for when `dummyData[i]["isSent"]` is `false`? Does that mean `false == null`? or should this be `if(dummyData[i]["isSent"] != false`?

Comment (HelenGezahegn) [reply lib/chat/user_message_rows.dart:132]: Oh, good catch! I definitely wasn't paying attention there -- thanks! I'll fix after work. 👍

File: lib/profile/sections/skills.dart [hunk 2]
```diff
@@ -25,9 +25,9 @@ class ProfileSkills extends StatelessWidget {
   }
 
   buildSkillList(BuildContext context) {
-    List<Widget> expertList = new List<Widget>();
-    List<Widget> intermediateList = new List<Widget>();
-    List<Widget> beginnerList = new List<Widget>();
+    var expertList = <Widget>[];
+    var intermediateList = <Widget>[];
+    var beginnerList = <Widget>[];
```
[the remark this answers is not in the document]
Comment (xiangxinyue) [reply lib/profile/sections/skills.dart:30]: Great point!

[the remark this answers is not in the document]
Comment (HelenGezahegn) [reply lib/profile/sections/skills.dart]: Oof, I looked into some other `effective_dart` lint errors and the `omit_local_variable_types` tells me to change these back to `var`. We'll deal with that when we get to that PR though. 👍

**Pull request merged**